common/crypto.hpp

```diff
@@ -37,6 +37,22 @@ enum CryptoErrorCode {
 	VerificationError,
 };
 
+class PrivateKey;
+using ExpectedPrivateKey = expected::expected<PrivateKey, error::Error>;
+
+class PrivateKey {
+public:
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path, const string &passphrase);
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path);
+#ifdef MENDER_CRYPTO_OPENSSL
+	unique_ptr<EVP_PKEY, void (*)(EVP_PKEY *)> pkey;
+
+private:
+	PrivateKey(unique_ptr<EVP_PKEY, void (*)(EVP_PKEY *)> &private_key) : pkey(std::move(private_key)){};
```

Comment:
Observation:

A couple of things going on here.

1. Read about `move` constructors: https://en.cppreference.com/w/cpp/language/move_constructor

I guess this made it compile for you, but it looks a little odd. Unique_ptrs handle what you need in a sense.

Reply:
Indeed, `&private_key` should be `&&private_key, and `move` should not be used here, because the `&&` should already make it movable.

Reply:
This concept is totally new for me, and requires some practice on my hands. Please check if what I have now is better :)